currency/translation_test.go

```diff
@@ -34,3 +36,97 @@ func TestGetTranslation(t *testing.T) {
 		t.Errorf("received: '%v', but expected: '%v'", actual, XBT)
 	}
 }
+
+func TestNewTranslations(t *testing.T) {
+	t.Parallel()
+	translations := NewTranslations(map[Code]Code{
+		XBT:  BTC,
+		XETH: ETH,
+		XDG:  DOGE,
+		USDM: USD,
+	})
+	require.NotNil(t, translations)
+
+	if !translations.Translate(XBT).Equal(BTC) {
+		t.Error("NewTranslations: translation failed")
+	}
+
+	if !translations.Translate(LTC).Equal(LTC) {
+		t.Error("NewTranslations: translation failed")
+	}
+}
+
+func TestFindMatchingPairsBetween(t *testing.T) {
+	t.Parallel()
+	ltcusd := NewPair(LTC, USD)
+
+	spotPairs := Pairs{
+		NewPair(BTC, USD),
+		NewPair(ETH, USD),
+		NewPair(ETH, BTC),
+		ltcusd,
+	}
+
+	futuresPairs := Pairs{
+		NewPair(XBT, USDM),
+		NewPair(XETH, USDM),
+		NewPair(XETH, BTCM),
+		ltcusd,             // exact match
+		NewPair(XRP, USDM), // no match
+	}
+
+	matchingPairs := FindMatchingPairsBetween(PairsWithTranslation{spotPairs, nil}, PairsWithTranslation{futuresPairs, nil})
+	require.Len(t, matchingPairs, 1)
+
+	if !matchingPairs[ltcusd].Equal(ltcusd) {
+		t.Error("FindMatchingPairsBetween: matching pair not found")
+	}
+
+	translations := NewTranslations(map[Code]Code{
+		XBT:  BTC,
+		XETH: ETH,
+		XDG:  DOGE,
+		USDM: USD,
+		BTCM: BTC,
+	})
+
+	matchingPairs = FindMatchingPairsBetween(PairsWithTranslation{spotPairs, nil}, PairsWithTranslation{futuresPairs, translations})
```

Comment:
I feel this test could be improved upon by actually using the `matchingPairs` response to ensure that all expected pairs from `futuresPairs` is contained instead of a len check. Could also solidify your design of this translator by adding a scenario with delimiters to show that you can translate between different one, so then the test will fail if someone ever attempts to change that - stronger than a comment above the function